oid4vc/oid4vc/models/dcql_query.py

```diff
@@ -0,0 +1,412 @@
+                ):  # TODO: check logic here
+                    raise ValidationError("Path elements must be string, int, or None.")
+
+        values = data.get("values")
+        if values:
+            for v in values:
+                if not (
+                    isinstance(v, str) or isinstance(v, int) or isinstance(v, bool)
+                ):  # TODO: check logic here
+                    raise ValidationError(
+                        "Values elements must be string, int, or bool."
+                    )
+
+
+ClaimQueryID = str
+
+
+class CredentialMeta(BaseModel):
+    """Metadata for credential query."""
+
+    class Meta:
+        """Credential Meta Schema Metadata."""
+
+        schema_class = "CredentialMetaSchema"
+
+    def __init__(
+        self,
+        query_type: Optional[str] = None,
+        doctype_values: Optional[List[str]] = None,
+        vct_values: Optional[List[str]] = None,
+    ):
+        """Initialize a CredentialMeta model."""
+        super().__init__()
+
+        self.query_type = query_type
+        self.doctype_values = doctype_values
+        self.vct_values = vct_values
+
+
+class CredentialMetaSchema(BaseModelSchema):
+    """Schema for credential query metadata."""
+
+    class Meta:
+        """Credential Meta Schema Metadata."""
+
+        model_class = "CredentialMeta"
+
+    query_type = fields.Str(
+        required=False,
+        validate=OneOf(
+            [
+                "SdJwtVcMeta",
+                "MdocMeta",
+            ]
+        ),
+    )
+
+    doctype_values = fields.List(
+        fields.Str,
+        required=False,
+    )
+
+    vct_values = fields.List(
+        fields.Str,
+        required=False,
+    )
+
+    @validates_schema
+    def validate_fields(self, data, **kwargs):
+        """Validate CredentialMeta object."""
+
+        query_type = data.get("query_type")
+        doctype_values = data.get("doctype_values")
+        vct_values = data.get("vct_values")
+
+        if query_type == "SdJwtVcMeta" and doctype_values:
+            raise ValidationError("SdJwtVc Meta cannot contain `doctype_values`")
+
+        if query_type == "MdocMeta" and vct_values:
+            raise ValidationError("Mdoc Meta cannot contain `vct_values`")
+
+        if vct_values and doctype_values:
+            raise ValidationError(
+                "Credential Metadata cannot have both vct_values and doctype_values."
+            )
+
+
+class CredentialQuery(BaseModel):
+    """A DCQL Credential Query."""
+
+    class Meta:
+        """Credential Query Metadata."""
+
+        schema_class = "CredentialQuerySchema"
+
+    def __init__(
+        self,
+        *,
+        credential_query_id: str,
+        format: str,
+        meta: Optional[CredentialMeta] = None,
+        claims: Optional[List[ClaimsQuery]] = None,  # min_length of 1
+        claim_sets: Optional[List[List[ClaimQueryID]]] = None,  # min_length of 1
```

Comment:
We should add a field validator to enforce these minimum length requirements when claims or claim sets are not None